packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -0,0 +1,512 @@
+   *   "host": <required: instance host name>,
+   *   "username": <required: username>,
+   *   "password": <required: password>,
+   *   "dbname": <optional: database name>,
+   *   "port": <optional: if not specified, default port will be used>,
+   *   "masterarn": <required: the arn of the master secret which will be used to create users/change passwords>
+   * }
+   * ```
+   */
+  readonly secret: secretsmanager.ISecret;
+
+  /**
+   * Specifies the number of days after the previous rotation before
+   * Secrets Manager triggers the next automatic rotation.
+   *
+   * @default Duration.days(30)
+   */
+  readonly automaticallyAfter?: Duration;
+}
+
+/**
+ * Create a Redshift Cluster with a given number of nodes.
+ * Implemented by {@link Cluster} via {@link ClusterBase}.
+ */
+export interface ICluster extends IResource, ec2.IConnectable, secretsmanager.ISecretAttachmentTarget {
+  /**
+   * Name of the cluster
+   *
+   * @attribute ClusterName
+   */
+  readonly clusterName: string;
+
+  /**
+   * The endpoint to use for read/write operations
+   *
+   * @attribute EndpointAddress,EndpointPort
+   */
+  readonly clusterEndpoint: Endpoint;
+}
+
+/**
+ * Properties that describe an existing cluster instance
+ */
+export interface ClusterAttributes {
+  /**
+   * The security groups of the redshift cluster
+   *
+   * @default no security groups will be attached to the import
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * Identifier for the cluster
+   */
+  readonly clusterName: string;
+
+  /**
+   * Cluster endpoint address
+   */
+  readonly clusterEndpointAddress: string;
+
+  /**
+   * Cluster endpoint port
+   */
+  readonly clusterEndpointPort: number;
+
+}
+
+/**
+ * Properties for a new database cluster
+ */
+export interface ClusterProps {
+
+  /**
+   * An optional identifier for the cluster
+   *
+   * @default - A name is automatically generated.
+   */
+  readonly clusterIdentifier?: string;
+
+  /**
+   * Additional parameters to pass to the database engine
+   * https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html
+   *
+   * @default - No parameter group.
+   */
+  readonly parameterGroup?: IClusterParameterGroup;
+
+  /**
+   * Number of compute nodes in the cluster
+   *
+   * Value must be at least 1 and no more than 100.
+   *
+   * @default 1
+   */
+  readonly numberOfNodes?: number;
+
+  /**
+   * The node type to be provisioned for the cluster.
+   *
+   * @default {@link NodeType.DC2_LARGE}
+   */
+  readonly nodeType?: NodeType;
+
+  /**
+   * Settings for the individual instances that are launched
+   *
+   * @default {@link ClusterType.MULTI_NODE}
+   */
+  readonly clusterType?: ClusterType;
+
+  /**
+   * What port to listen on
+   *
+   * @default - The default for the engine is used.
+   */
+  readonly port?: number;
+
+  /**
+   * Whether to enable encryption of data in the cluster
+   *
+   * @default false
+   */
+  readonly encrypted?: boolean
```

Comment:
How is this property related to encryptionKey? Can we make it depend on whether the encryption Key was passed or not?

Reply:
> How is this property related to encryptionKey?

If `encrypted` is true and no `encryptionKey` is set, then the cluster is encrypted using an AWS-managed key. I worked a little on the docstring. Please let me know, if it still needs work.

> Can we make it depend on whether the encryption Key was passed or not?

I would rather not. Using the AWS-managed key for encryption at rest is sufficient, cheaper and leaves less space for error. Imho (using) a CMK has more room for mistakes. Do you disagree?

Reply:
If that is the case, how do you feel about making `encrypted` `true` by default? Seems like it doesn't have much downside... or am I missing something?

Reply:
Also, what if `encrypted` is `false`, and `encryptionKey` has been provided? I guess `encryptionKey` will be ignored in that case? Perhaps we should fail with a validation error if that happens?